src/SimpleCalculator.Web/Controllers/HomeController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+using SimpleCalculator.Web.Models;
+using Calculations;
+
+namespace SimpleCalculator.Web.Controllers;
+
+public class HomeController : Controller
+{
+    private readonly IAddition _addition;
+    private readonly ISubtraction _subtraction;
+    private readonly IMultiplication _multiplication;
+
+    public HomeController(IAddition addition, ISubtraction subtraction, IMultiplication multiplication)
+    {
+        _addition = addition;
+        _subtraction = subtraction;
+        _multiplication = multiplication;
+    }
+
+    public IActionResult Index()
+    {
+        return View();
+    }
+
+    [HttpPost]
+    public IActionResult Add(CalculationInputModel model)
+    {
+        model.Result = _addition.Add(model.FirstNumber, model.SecondNumber);
+        ViewData["Result"] = model.Result;
+        return View("Index");
```

Comment:
Why you wouldn't pass `model` to View? You don't need to use ViewData. Here is an article about ViewData https://www.dotnettricks.com/learn/mvc/viewdata-vs-viewbag-vs-tempdata-vs-session

Reply:
Thanks I will take a look at it.